blocks/file-blocks/flat.tsx

```diff
@@ -43,3 +29,51 @@ export default function (props: FileBlockProps) {
     </div>
   );
 }
+
+function parseCSV(content: string): any[] {
+  try {
+    const rows = csvParseRows(content);
+    const headers = rows[0];
+    const csvData = rows.slice(1).map((row: any) => {
+      return headers.reduce((acc: Record<string, any>, key, i) => {
+        acc[key] = Number.isFinite(row[i]) ? +row[i] : row[i];
+        return acc;
+      }, {});
+    });
+    return csvData;
+  } catch (e) {
+    return [];
+  }
+}
+function parseYML(content: string): any[] {
+  try {
+    const data = load(content);
+    const isArray = Array.isArray(data);
+    if (isArray) return data;
+    // can we flatten an object into something sensical?
+    const flattenedData = Object.keys(data).map((key) => {
+      const contents = data[key];
+      const expandedContents =
+        typeof contents === "object" && contents !== null
+          ? contents
+          : { contents: JSON.stringify(contents) };
+      return {
+        __KEY__: key,
+        ...expandedContents,
+      };
+    });
+    return flattenedData;
```

Comment:
the idea here is to return a table of the object properties? does `__KEY__` do something magical in `flat-ui`?

Reply:
oic, nice